app/services/open-weather.ts

```diff
@@ -0,0 +1,51 @@
+import Service from '@ember/service';
+import fetch from 'fetch';
+
+import config from 'weather-app/config/environment';
+
+export default class OpenWeatherService extends Service {
+  fetchCurrentWeather(coord: Coordinate) {
+    return this.fetchWeather('weather', coord);
+  }
+
+  fetchForecast(coord: Coordinate) {
+    return this.fetchWeather('forecast', coord);
+  }
+
+  fetchWeather(type: string, coord: Coordinate): any {
+    const parseJSON = (response: any) => {
+      return response.json();
+    };
+
+    return fetch(`https://${config.APP.rapid.host}/${type}?lat=${coord.latitude}&lon=${coord.longitude}&units=metric`, {
+      method: 'GET',
+      headers: {
+        'Content-Type': 'application/json',
+        'X-RapidAPI-Host': config.APP.rapid.host,
+        'X-RapidAPI-Key': config.APP.rapid.apiKey,
+      },
+    }).then(parseJSON)
+    .then(this.normalizeResponse);
+  }
+
+  normalizeResponse(response: any): OpenWeatherData {
+    const name = `${response.name}, ${response.sys.country}`;
+    const details = {
```

Comment:
just FYI and it is totally a style thing but you could make use of two ES6 features here.

1 you can destructure out the props you care about
2 as the props have the same name of the object you do not need to do `humidity: humidity` just `humidity`

for example:

const {humidity, pressure, sunrise, sunset, temp_max, temp_min} = response.main;
const details = {
   humidity,
   pressure
   ... and so on
}
```

the problem is you also have things from sys and wind I see so maybe actually it is not worth it after all, if you want all from main sys and wind though you could use spreading to do the same thing like this

```
const details = {
  ...response.main
  ...response.sys
  ...response.wind
}
```

but yeah only if you want all of them 😓 

tbh I am just thinking out loud at this stage 🤣

Reply:
Yes, thanks! I need all the values from main but only a few from sys and wind. So, I am using a combination of de-structuring and spreading.